File: abstractions/java/lib/src/main/java/com/microsoft/kiota/store/BackedModel.java
package com.microsoft.kiota.store;

import javax.annotation.Nonnull;

/** Defines the contracts for a model that is backed by a store. */
public interface BackedModel {
    /**
     * Gets the store that is backing the model.
     * @return the backing store.
     */
    @Nonnull
    BackingStore getBackingStore();
}

File: abstractions/php/src/Store/BackedModel.php
<?php


namespace Microsoft\Kiota\Abstractions\Store;

/** Defines the contracts for a model that is backed by a store. */
interface BackedModel {
    /**
     * Gets the store that is backing the model.
     * @return BackingStore|null the backing store.
     */
    public function getBackingStore(): ?BackingStore;
}
